Code/Framework/AzToolsFramework/Tests/UI/AssetBrowserTests.cpp

```diff
@@ -0,0 +1,392 @@
+        D:
+          \\
+            dev
+              o3de
+                GameProject
+                  Assets
+                    Source_1
+                      Product_1_1
+                      Product_1_0
+                    Source_0
+                      Product_0_3
+                      Product_0_2
+                      Product_0_1
+                      Product_0_0
+                  Scripts
+                    Source_3
+                    Source_2
+                      Product_2_2
+                      Product_2_1
+                      Product_2_0
+                  Misc
+                    Source_4
+                      Product_4_2
+                      Product_4_1
+                      Product_4_0 )";
+
+        [[maybe_unused]]AZStd::vector<QString> hierarchySections = GetVectorFromFormattedString(testHierarchy);
+
+        namespace AzAssetBrowser = AzToolsFramework::AssetBrowser;
+
+        AddScanFolder(m_folderIds.at(2), "D:/dev/o3de/GameProject/Misc", "Misc");
+        AZ::Uuid sourceUuid_4 = CreateSourceEntry(m_sourceIDs.at(4), m_folderIds.at(2), "Source_4");
+        CreateProduct(m_productIDs.at(0), sourceUuid_4, "Product_4_0");
+        CreateProduct(m_productIDs.at(1), sourceUuid_4, "Product_4_1");
+        CreateProduct(m_productIDs.at(2), sourceUuid_4, "Product_4_2");
+
+        AddScanFolder(m_folderIds.at(1), "D:/dev/o3de/GameProject/Scripts", "Scripts");
+
+        AZ::Uuid sourceUuid_2 = CreateSourceEntry(m_sourceIDs.at(2), m_folderIds.at(1), "Source_2");
+        CreateProduct(m_productIDs.at(0), sourceUuid_2, "Product_2_0");
+        CreateProduct(m_productIDs.at(1), sourceUuid_2, "Product_2_1");
+        CreateProduct(m_productIDs.at(2), sourceUuid_2, "Product_2_2");
+
+        CreateSourceEntry(m_sourceIDs.at(3), m_folderIds.at(1), "Source_3");
+
+        AddScanFolder(m_folderIds.at(0), "D:/dev/o3de/GameProject/Assets", "Assets");
+
+        AZ::Uuid sourceUuid_0 = CreateSourceEntry(m_sourceIDs.at(0), m_folderIds.at(0), "Source_0");
+        CreateProduct(m_productIDs.at(0), sourceUuid_0, "Product_0_0");
+        CreateProduct(m_productIDs.at(1), sourceUuid_0, "Product_0_1");
+        CreateProduct(m_productIDs.at(2), sourceUuid_0, "Product_0_2");
+        CreateProduct(m_productIDs.at(3), sourceUuid_0, "Product_0_3");
+
+        AZ::Uuid sourceUuid_1 = CreateSourceEntry(m_sourceIDs.at(1), m_folderIds.at(0), "Source_1");
+        CreateProduct(m_productIDs.at(0), sourceUuid_1, "Product_1_0");
+        CreateProduct(m_productIDs.at(1), sourceUuid_1, "Product_1_1");
+
+        QString resultMessage = QString();
+        AZStd::function<void(const QString&)> printer = [&resultMessage](const QString& message)
+        {
+            resultMessage += message + "\n";
+        };
+        PrintModel(m_assetBrowserComponent->GetAssetBrowserModel(), printer);
+        [[maybe_unused]] AZStd::vector<QString> createdHierarchySections = GetVectorFromFormattedString(resultMessage);
```

Comment:
question: I take it the plan is to do something useful with this in a follow-up PR?

Reply:
Yes! That's the plan